lib/internal/webstreams/adapters.js

```diff
@@ -407,7 +413,12 @@ function newReadableStreamFromStreamReadable(streamReadable) {
   streamReadable.on('data', onData);
 
   return new ReadableStream({
-    start(c) { controller = c; },
+    async start(c) {
+      controller = c;
+      if (!streamReadable._readableState.constructed) {
+        await EE.once(kConstruct, streamReadable);
```

Comment:
Isn't this backward?
```suggestion
        await EE.once(streamReadable, kConstruct);
```